step-core/src/main/java/step/attachments/FileResolver.java

```diff
@@ -1,120 +1,154 @@
-		}
-
-		public File getFile() {
-			return file;
-		}
-
-		@Override
-		public void close() throws IOException {
-			if(resourceRevisionFileHandle!=null) {
-				resourceRevisionFileHandle.close();
-			}
-		}
-	}
-}
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *  
+ * This file is part of STEP
+ *  
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *  
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *  
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.attachments;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+
+import step.resources.Resource;
+import step.resources.ResourceManager;
+import step.resources.ResourceRevisionFileHandle;
+
+public class FileResolver {
+
+	public static final String ATTACHMENT_PREFIX = "attachment:";
+	public static final String RESOURCE_PREFIX = "resource:";
+	public static final String RESOURCE_PATH_SEPARATOR = ":";
+	
+	private final ResourceManager resourceManager;
+	
+	public FileResolver(ResourceManager resourceManager) {
+		super();
+		this.resourceManager = resourceManager;
+	}
+
+	public ResourceManager getResourceManager() {
+		return resourceManager;
+	}
+
+	public File resolve(String path) {
+		File file;
+		if(path.startsWith(ATTACHMENT_PREFIX)) {
+			throw new RuntimeException("Attachments have been migrated to the ResourceManager. The reference " + path +
+					" isn't valid anymore. Your attachment should be migrated to the ResourceManager.");
+		} else if(path.startsWith(RESOURCE_PREFIX)) {
+			file = getResourceRevisionFileHandleForPath(path).getResourceFile();
+		} else {
+			file = new File(path);
+		}
+		return file;
+	}
+	
+	public static String resolveResourceId(String path) {
+		String resourceId;
+		if(path != null && path.startsWith(RESOURCE_PREFIX)) {
+			String subResourcePath = extractResourceSubPath(path);
+			resourceId = subResourcePath.split(RESOURCE_PATH_SEPARATOR)[0];
+		} else {
+			resourceId = null;
+		}
+		return resourceId;
+	}
+
+	public static String resolveRevisionId(String path) {
+		String revisionId = null;
+		if(path != null && path.startsWith(RESOURCE_PREFIX)) {
+			String subResourcePath = extractResourceSubPath(path);
+			String[] split = subResourcePath.split(RESOURCE_PATH_SEPARATOR);
+			if (split.length == 2){
```

Comment:
TO avoid unexpected null values, we should throw a clear error if we expect a revision to be present and don't have one